### ISSUE TITLE
warning: URI.escape is obsolete

### ISSUE THREAD

**Issue by lilith** [chargebee/chargebee-ruby#23]
warning: URI.escape is obsolete

```
ruby 2.1.8

chargebee-2.2.7/lib/chargebee/models/model.rb:59:in `block in uri_path': warning: URI.escape is obsolete

```

**Comment by collimarco** [comment]
I see this warning on **Ruby 2.7** (I didn't see it on <= 2.6).

It completely pollutes all the output of `rspec` and logs... 

**Please fix this!**

More info:
https://docs.knapsackpro.com/2020/uri-escape-is-obsolete-percent-encoding-your-query-string

**Comment by evolve2k** [comment]
This error is crashing my app when I deploy it to heroku.
I've just upgraded my app to ruby 2.7

Bump. Please resolve.

**Comment by remy727** [comment]
Same here. I am using `ruby 2.7.2`

**Comment by blgastesi** [comment]
It fails on `ruby 3.0`

```undefined method `encode' for URI:Module```

**Comment by collimarco** [comment]
To everyone involved in this conversation, **please contact Chargebee support with email / ticket so that we can raise the priority of this issue!**

It's a serious issue since upgrades to new Ruby versions (2.7+) cause *thousands* of warnings in tests, logs, etc. or even worse... in the latest Ruby version (3) break the application completely!

It's just a single method to fix... how is it possible that this issue is still open after long time!?

**Comment by ryansdwilson** [comment]
I just opened a ticket. I can't believe this isn't fixed after so much time. @1cb-navaneedhan this is worth fixing!

**Comment by ryansdwilson** [comment]
Hey @cb-goutham have you thought about fixing this issue?

**Comment by cb-rakesh** [comment]
This has been fixed and released in v2.8.7.
